core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -274,7 +275,7 @@ CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> p
     // Visible for testing.
     SharePartition sharePartition(ShareFetchPartitionData shareFetchPartitionData, TopicIdPartition topicIdPartition) {
         try {
-            Persister persister = null;
+            Persister persister = new NoOpShareStatePersister();
```

Comment:
1. Do we require to instantiate `NoOpShareStatePersister` when later an actual implementation can be provided, I think no.
2. Do we require to create a new instance of `NoOpShareStatePersister` for each SharePartition, again I think no as `NoOpShareStatePersister` can be Singleton. It will be good to have an instacne exposed from `NoOpShareStatePersister` which can be used in all SharePartition.

Reply:
makes sense, I'll convert the `NoOpShareStatePersister` to singleton class and use the `getInstance` method here and in the tests as well.